File: pat/advance/pat1029.java
import java.util.*;

public class pat1029
{
    public static void main(String[] args)
    {
        Scanner in = new Scanner(pat1029.class.getClassLoader().getResourceAsStream("test"));
        
        int n = in.nextInt();
        int[] a = new int[n];
        for(int i=0;i<n;++i)
            a[i] = in.nextInt();

        int m = in.nextInt();
        int[] b = new int[m];
        for(int i=0;i<m;++i)
            b[i] = in.nextInt();

        if(m==1 && n==1)
            System.out.print(a[0]>b[0]?b[0]:a[0]);
        else
        {
            int k = (n+m+1)/2;
            int i = 0; //present position, visited
            int j = 0;
            int c = 0;
            int median = 0;
            while(c<k && i<n && j<m)
            {
                if(a[i]<b[j])
                {
                    median = a[i];
                    i++;
                }
                else
                {
                    median = b[j];
                    j++;
                }

                c++;
            }

            if(c==k)
                System.out.print(median);
            else
            {
                if(i==n)
                {
                    while(c<k)
                    {
                        j++;
                        c++;
                    }
                        
                    System.out.print(b[j]);
                }
                else
                {
                    while(c<k)
                    {
                        i++;
                        c++;
                    }
                    System.out.print(a[i]);
                }
            }

        }
    }
}

File: pat/advance/pat1029.py
a=[]
b=[]
f = open("test", "rU")
for x in f.readline().split():
    a.append(int(x))
for x in f.readline().split():
    b.append(int(x))

n = a[0]
m = b[0]
k = (m+n+1)/2
c = 0
i=1
j=1
while c<k and i<=n and j<=m:
    if a[i]<b[j]:
        median = a[i]
        i+=1
    else:
        median = b[j]
        j+=1
    c+=1

if c==k:
    print median
else:
    if i>n:
        while c<k:
            j+=1
            c+=1
        print b[j]
    else:
        while c<k:
            i+=1
            c+=1
        print a[i]
f.close()
print '1234'[1:-1]